src/main.rs

```diff
@@ -27,35 +28,36 @@ run()
 
 fn main() {
     // init device
-    let device_handle = devices::tactrix_openport::new();
+    let device_handle = Arc::new(devices::tactrix_openport::new());
     // init server
     let server = std::net::TcpListener::bind("127.0.0.1:9001").unwrap();
     // listen for connections
-    let websockets = Arc::new(Mutex::new(vec![]));
+    let mut websockets = vec![];
     for stream in server.incoming() {
-        let mut websocket = tungstenite::server::accept(stream.unwrap()).unwrap();
-        websockets.lock().unwrap().push(websocket);
+        let websocket = Arc::new(Mutex::new(tungstenite::server::accept(stream.unwrap()).unwrap()));
```

Comment:
Same explanation as the device handle, but we need a mutable reference to the websocket to call its methods (so we need to add mutex)